src/metaopt/core/io/space_builder.py

```diff
@@ -206,9 +234,172 @@ def build(self, name, expression):
         return dimension
 
 
-def build():
-    """Create a definition of the problem's search space, using information
-    from the user's script configuration and arguments.
+class SpaceBuilder(object):
+    """Build a `Space` object form user's configuration."""
 
-    """
-    pass
+    USERCONFIG_OPTION = '--config='
+    USERCONFIG_KEYWORD = 'mopt~'
+    USERARGS_SEARCH = r'\W*([a-zA-Z0-9_-]+)~(.*)'
+    USERARGS_TMPL = r'(.*)~(.*)'
+
+    def __init__(self):
+        """Initialize a `SpaceBuilder`."""
+        self.userconfig = None
+        self.is_userconfig_an_option = None
+        self.userargs_tmpl = None
+        self.userconfig_tmpl = None
+        self.dimbuilder = DimensionBuilder()
+        self.space = None
+        self.converter = None
+
+    def build_from(self, cmd_args):
+        """Create a definition of the problem's search space, using information
+        from the user's script configuration (if provided) and command line arguments.
+
+        :param cmd_args: A list of command line arguments provided for the user's script.
+
+        .. note:: A template configuration file complementing user's script can be
+           provided either by explicitly using the prefix '--config=' or by being the
+           first positional argument.
+
+        """
+        self.userargs_tmpl = None
+        self.userconfig_tmpl = None
+        self.space = Space()
+
+        self.userconfig, self.is_userconfig_an_option = self._build_from_args(cmd_args)
+
+        if self.userconfig:
+            self._build_from_config(self.userconfig)
+
+        log.debug("Configuration and command line arguments were parsed and "
+                  "a `Space` object was built successfully:\n%s", self.space)
+
+        return self.space
+
+    def _build_from_config(self, config_path):
```

Comment:
Will we need a build_from_config like this in many classes like experiment, database, space? It seems to me that much of the loop could be encapsulated inside the parser. If we could for instance convert other config file types to YAML we could always instantiate using yaml mechanisms.